kotlin/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.kt

```diff
@@ -0,0 +1,216 @@
+/* Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.viewshedgeoprocessing
+
+import android.graphics.Color
+import android.os.Bundle
+import android.view.MotionEvent
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.concurrent.Job
+import com.esri.arcgisruntime.concurrent.ListenableFuture
+import com.esri.arcgisruntime.data.FeatureCollectionTable
+import com.esri.arcgisruntime.data.Field
+import com.esri.arcgisruntime.geometry.GeometryType
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.mapping.view.Graphic
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol
+import com.esri.arcgisruntime.symbology.SimpleMarkerSymbol
+import com.esri.arcgisruntime.symbology.SimpleRenderer
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingFeatures
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingJob
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingParameters
+import com.esri.arcgisruntime.tasks.geoprocessing.GeoprocessingTask
+import kotlinx.android.synthetic.main.activity_main.*
+import java.util.concurrent.ExecutionException
+import kotlin.math.roundToInt
+
+class MainActivity : AppCompatActivity() {
+
+  private val geoprocessingTask: GeoprocessingTask by lazy { GeoprocessingTask(getString(R.string.viewshed_service)) }
+  private var geoprocessingJob: GeoprocessingJob? = null
+
+  private val inputGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+  private val resultGraphicsOverlay: GraphicsOverlay by lazy { GraphicsOverlay() }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a map with the Basemap type topographic
+    val map = ArcGISMap(Basemap.Type.TOPOGRAPHIC, 45.3790902612337, 6.84905317262762, 12)
+    // set the map to be displayed in this view
+    mapView.map = map
```

Comment:
Could combine this with the touch listener (and maybe the graphicsoverlays? --your call on that). Something like:
```
mapView.apply {
map = Map stuff

onTouchListener = DefaultMapViewOnTouchListener

graphicsOverlay.add(resultGraphicsOverlay)
graphicsOverlay.add(inputGraphicsOverlay)
}
```

Reply:
good call!